service/src/main/java/bio/terra/pipelines/common/utils/PipelineInputTypesEnum.java

```diff
@@ -0,0 +1,110 @@
+package bio.terra.pipelines.common.utils;
+
+import bio.terra.common.exception.ValidationException;
+import java.util.List;
+
+public enum PipelineInputTypesEnum {
+  STRING {
+    @Override
+    public String cast(String fieldName, Object value) {
+      validateNotNullOrEmpty(fieldName, value);
+      if (value instanceof String stringValue) {
+        return stringValue.trim();
+      } else {
+        throw new ValidationException("%s must be a string".formatted(fieldName));
+      }
+    }
+  },
+  INTEGER {
+    @Override
+    public Integer cast(String fieldName, Object value) {
+      validateNotNullOrEmpty(fieldName, value);
+      if (value instanceof Integer integerValue) {
+        return integerValue;
+      } else if (value instanceof String stringValue) {
+        try {
+          return Integer.parseInt(stringValue);
+        } catch (NumberFormatException e) {
+          throw new ValidationException("%s must be an integer".formatted(fieldName));
+        }
+      } else {
+        throw new ValidationException("%s must be an integer".formatted(fieldName));
+      }
+    }
+  },
+  VCF {
+    @Override
+    public String cast(String fieldName, Object value) {
+      validateNotNullOrEmpty(fieldName, value);
+      if (value instanceof String stringValue) {
```

Comment:
is there a reason we dont use the `STRING` cast function here instead of duplicating it?  Would help keep things consistent if we could

Reply:
ah yeah I thought about that at some point but didn't do it. great suggestion!